core/unit_test/TestExecSpacePartitioning.hpp

```diff
@@ -131,9 +131,8 @@ TEST(TEST_CATEGORY, partitioning_by_args) {
 }
 
 TEST(TEST_CATEGORY, partitioning_by_vector) {
-  std::vector<int> weights{1, 1};
-  auto instances =
-      Kokkos::Experimental::partition_space(TEST_EXECSPACE(), weights);
+  auto instances = Kokkos::Experimental::partition_space(
+      TEST_EXECSPACE(), std::vector<int> /*weights*/ {1, 1});
```

Comment:
It does not make it obvious to the reader that you mean to call with a temporary.
Consider doing
```C++
static_assert(
  std::is_invocablle_v<
    decltype(Kokkos::Experimental::partition_space<int>),
    std::vector<int>&&>,
  "intended to be callable with temporary"
);
```

Reply:
I would need overload resolution for `Kokkos::Experimental::partition_space<ExecSpace, int>)` since both overloads are eligible. That means that I would need to simulate the call before testing that the function is callable with the arguments I need to provide for overload resolution which seems to defeat the purpose.

I just added a comment making the intent clear.